pycqed/measurement/measurement_control.py

```diff
@@ -666,22 +890,33 @@ def update_plotmon(self, force_update=False):
                 self._mon_upd_time = time.time()
                 time_since_last_mon_update = 1e9
             try:
+                cf = self.exp_metadata.get("compression_factor", 1)
                 if (time_since_last_mon_update > self.plotting_interval() or
                         force_update):
 
                     nr_sweep_funcs = len(self.sweep_function_names)
-                    for y_ind in range(len(self.detector_function.value_names)):
-                        for x_ind in range(nr_sweep_funcs):
-                            x = self.dset[:, x_ind]
-                            y = self.dset[:, nr_sweep_funcs+y_ind]
+                    for y_ind, vn in enumerate(
```

Comment:
Here (lines 898-911) it might also be useful to add short comments before blocks of code explaining what is done.